File: output/ChatGPT/avatar/Java/Python/atcoder_ABC143_D.py
def find(li, from_index, target):
    low = from_index
    upp = len(li) - 1
    mid = (low + upp) // 2 + 1 if (upp - low + 1) % 2 == 0 else (low + upp) // 2

    if upp - low < 0:
        return 0
    elif li[from_index] >= target:
        return 0
    elif li[upp] < target:
        return upp - low + 1

    while upp - low > 1:
        if li[mid] >= target:
            upp = mid
        else:
            low = mid
        mid = (low + upp) // 2 + 1 if (upp - low + 1) % 2 == 0 else (low + upp) // 2

    return low - from_index + 1

N = int(input())
L = list(map(int, input().split()))
L.sort()
count = 0
for i in range(N):
    for j in range(i + 1, N):
        a = L[i]
        b = L[j]
        res = find(L, j + 1, a + b)
        count += res

print(count)

File: output/ChatGPT/avatar/Java/C++/atcoder_ABC143_D.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int find(vector<int>& li, int from, int target) {
    int low = from;
    int upp = li.size() - 1;
    int mid = (upp - low + 1) % 2 == 0 ? (low + upp) / 2 + 1 : (low + upp) / 2;
    if (upp - low < 0) {
        return 0;
    }
    else if (li[from] >= target) {
        return 0;
    }
    else if (li[upp] < target) {
        return upp - low + 1;
    }
    while (upp - low > 1) {
        if (li[mid] >= target) {
            upp = mid;
        }
        else {
            low = mid;
        }
        mid = (upp - low + 1) % 2 == 0 ? (low + upp) / 2 + 1 : (low + upp) / 2;
    }
    return low - from + 1;
}

int main() {
    int N;
    cin >> N;
    vector<int> L(N);
    for (int i = 0; i < N; i++) {
        cin >> L[i];
    }
    sort(L.begin(), L.end());
    int count = 0;
    for (int i = 0; i < N; i++) {
        for (int j = i + 1; j < N; j++) {
            int a = L[i];
            int b = L[j];
            int res = find(L, j + 1, a + b);
            count += res;
        }
    }
    cout << count << endl;

    return 0;
}